src/main/java/org/mule/service/scheduler/internal/AbstractRunnableFutureDecorator.java

```diff
@@ -110,27 +129,38 @@ protected void doRun(RunnableFuture<V> task) {
     }
 
     try {
+      if (profilingService != null) {
```

Comment:
wrap this functionality in methods to avoid duplicated code.

Reply:
Functional sugar degrades performance, so I assume that you are referring to the **_profilingService != null_** fragment